pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```
@@ -243,75 +237,29 @@ private Map<String, Map<String, String>> reassignSegments(String instancePartiti
-    Map<String, SegmentZKMetadata> segmentZKMetadataMap = new HashMap<>();
-    for (SegmentZKMetadata segmentZKMetadata : segmentsZKMetadata) {
-      segmentZKMetadataMap.put(segmentZKMetadata.getSegmentName(), segmentZKMetadata);
-    }
-    int numPartitions = instancePartitions.getNumPartitions();
-    Map<Integer, List<String>> instancePartitionIdToSegmentsMap = new HashMap<>();
-    for (String segmentName : currentAssignment.keySet()) {
-      int partitionId = getPartitionId(segmentZKMetadataMap.get(segmentName));
-      int instancePartitionId = partitionId % numPartitions;
-      instancePartitionIdToSegmentsMap.computeIfAbsent(instancePartitionId, k -> new ArrayList<>()).add(segmentName);
-    }
-
-    // NOTE: Shuffle the segments within the current assignment to avoid moving only new segments to the new added
-    //       servers, which might cause hotspot servers because queries tend to hit the new segments. Use the table
-    //       name hash as the random seed for the shuffle so that the result is deterministic.
-    Random random = new Random(_offlineTableName.hashCode());
-    for (List<String> segments : instancePartitionIdToSegmentsMap.values()) {
-      Collections.shuffle(segments, random);
+  private void getSegmentAssignmentStrategy(Map<InstancePartitionsType, SegmentAssignmentConfig> segmentAssignmentConfigMap, int numReplicaGroups) {
+    if (segmentAssignmentConfigMap == null) {
+      // Set Default Segment Assignment Strategy in case not set
+      _segmentAssignmentStrategy = new OfflineNonReplicaGroupSegmentAssignmentStrategy();
+    } else {
+      String assignmentStrategy = segmentAssignmentConfigMap.get(InstancePartitionsType.OFFLINE).getAssignmentStrategy();
+      switch (assignmentStrategy) {
+        case AssignmentStrategy.REPLICA_GROUP_SEGMENT_ASSIGNMENT_STRATEGY:
+          Preconditions.checkState(numReplicaGroups != 1, "Not more than 1 replica group: %s" + _offlineTableName);
```

`numReplicaGroups = 1` is also valid case? 

The assignment will be the same as `BalacnedNumSegment` though.

---

right, so it will fall into default case.

---

@GSharayu I think that we will throw the exception here rather than fall back to the default case?